.golangci.yml

```diff
@@ -0,0 +1,110 @@
+# golangci configuration
+
+linters-settings:
+  dupl:
+    threshold: 100
+  funlen:
+    lines: 100
+    statements: 50
+  gci:
+    local-prefixes: github.com/kubevirt/kubevirt
+  goconst:
+    min-len: 2
+    min-occurrences: 2
+  gocritic:
+    enabled-tags:
+      - diagnostic
+      - experimental
+      - opinionated
+      - performance
+      - style
+    disabled-checks:
+      - dupImport # https://github.com/go-critic/go-critic/issues/845
+      - ifElseChain
+      - octalLiteral
+      - whyNoLint
+      - wrapperFunc
+    settings:
+      hugeParam:
+        sizeThreshold: 1024
+      rangeValCopy:
+        sizeThreshold: 1024
+  gocyclo:
+    min-complexity: 15
+  goimports:
+    local-prefixes: github.com/kiagnose/kiagnose
+  gomnd:
+    settings:
+      mnd:
+        # don't include the "operation" and "assign"
+        checks: argument,case,condition,return
+  govet:
+    check-shadowing: true
+  lll:
+    line-length: 140
+  maligned:
+    suggest-new: true
+  misspell:
+    locale: US
+  nolintlint:
+    allow-leading-space: true # don't require machine-readable nolint directives (i.e. with no leading space)
+    allow-unused: false # report any unused nolint directives
+    require-explanation: false # don't require an explanation for nolint directives
+    require-specific: false # don't require nolint directives to be specific about which linter is being skipped
+
+linters:
+  disable-all: true
+  enable:
+    - bodyclose
+    - deadcode
+    - depguard
+    - dogsled
+    - dupl
+    - errcheck
+    - exportloopref
+    - exhaustive
+    - funlen
+    - gochecknoinits
+    - goconst
+    - gocritic
+    - gocyclo
+    - gofmt
+    - goheader
+    - goimports
+    - gomnd
+    - goprintffuncname
+    - gosec
+    - gosimple
+    - govet
+    - ineffassign
+    - lll
+    - misspell
+    - nakedret
+    - noctx
+    - nolintlint
+    - rowserrcheck
+    - staticcheck
+    - structcheck
+    - stylecheck
+    - typecheck
+    - unconvert
+    - unparam
+    - unused
+    - varcheck
+    - whitespace
```

Comment:
why this particular list, and not any from the list below ?

Reply:
You are free to add, remove and update the configuration per need. This is just a base seen to work well on other projects.

Some of these projects are [kubernetes-nmstate](https://github.com/nmstate/kubernetes-nmstate/), [go-nft](https://github.com/networkplumbing/go-nft), [kiagnose](https://github.com/kiagnose/kiagnose/).
Perhaps @qinqon can share others.

Reply:
If it is not clear from my answer, I just copied it from the last project I worked on. And I guess me and others did the same with time, adjusting on each iteration per the experience of working on the project.